src/main/java/com/es/phoneshop/model/product/Product.java

```diff
@@ -25,6 +30,8 @@ public Product(Long id, String code, String description, BigDecimal price, Curre
         this.currency = currency;
         this.stock = stock;
         this.imageUrl = imageUrl;
+        this.priceHistory = new ArrayList<>();
+        priceHistory.add(new PriceDate(new Date(), price));
```

Comment:
priceHistory can be initialized in one line when you define it. Then you won't need to initialize it every time in constructor